moveit_core/cmake/moveit.cmake

```diff
@@ -0,0 +1,22 @@
+macro(moveit_build_options)
```

Comment:
I don't think that we want to migrate this. It seems to serve the exact same purpose as the moveit_package() macro and includes settings like the cxx14 standard that we definitely don't want to enable here. Since you already copied the build flags into the `moveit_package.cmake`, this file can be dropped imo. @tylerjw any thoughts?

Reply:
It makes sense to remove it. I forgot to delete this file while "redoing" this PR